### ISSUE TITLE
Archive option is not visible on some devices

### ISSUE THREAD

**Issue by naveensingh** [FossifyOrg/Messages#131]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 10

### Affected device model

Huawei P30 Pro

### How did you install the app?

None

### Steps to reproduce the bug

- Long press any conversation
- The **Archive** option isn't there.

### Expected behavior

The archive option should show up.

### Actual behavior

The archive option doesn't show up.

### Additional information

1. "Archived are supported by other apps such as Handcent or QKSMS." - The User.

2. The Archive option is hidden when the column doesn't exist:
https://github.com/FossifyOrg/Messages/blob/0d4ce785a5cc25e0e8b17552848602495d68c80d/app/src/main/kotlin/org/fossify/messages/extensions/Context.kt#L322-L328

**Comment by SynapseScribe** [comment]
Archive is not available on Huawei Y6 2019

**Comment by SynapseScribe** [comment]
i guess the archiving mechanism should be bound to the app and independent of OS

**Comment by naveensingh** [comment]
Yeah, that is one way to solve that problem. Unless more problems pop up, it'll be best to rely on the OS flag when possible and only fall back to the in-app DB when the column doesn't exist.
